### PR TITLE
Add support for cleanup function

### PR DESCRIPTION
**Pull request by badochov**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
https://github.com/phoenix-rtos/libphoenix/pull/313
https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/497
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: arch/ia32/crt0.S [hunk 1]
```diff
@@ -20,5 +20,6 @@
 .globl _start
 .type _start, %function
 _start:
-	jmp _startc
+	/* Call to ensure proper alignment. As %esp is 16 aligned at program entry. */
+	call _startc
```
Comment (agkaminski) [inline arch/ia32/crt0.S:24]: I don't get this. `call` does not align the stack in any way, it just pushes EIP (4 bytes). 16 byte alignment is a compiler requirement and not enforced by the CPU at all

Comment (badochov) [reply arch/ia32/crt0.S:24]: The call is here to ensure proper alignment in _startc, that is 16byte alignment prior to call. Should jmp be used _startc would be entered with rsp aligned to 16bytes, but it should be entered with (rsp -4) being 16 byte aligned

File: arch/riscv64/crt0.S [hunk 2]
```diff
@@ -23,12 +23,12 @@ _start:
 
 .option push
 .option norelax
-	la gp, __global_pointer$
+	la   gp, __global_pointer$
```
[the remark this answers is not in the document]
Comment (badochov) [reply arch/riscv64/crt0.S:26]: unfortunatelly clang doesnt support asm. OCD made me do it :stuck_out_tongue:

**Comment by badochov** [comment]
https://github.com/phoenix-rtos/phoenix-rtos-project/pull/969
test branch confirming it works with kernel changes

**Review by agkaminski — APPROVED** [review]

**Review by agkaminski — APPROVED** [review]

**Pull request merged**